app/controllers/security_group_controller.rb

```diff
@@ -79,18 +71,13 @@ def create_finished
     security_group_name = session[:async][:params][:name]
     task = MiqTask.find(task_id)
     if MiqTask.status_ok?(task.status)
-      add_flash(_("Security Group \"%{name}\" created") % { :name => security_group_name })
+      flash_and_redirect(_("Security Group \"%{name}\" created") % {:name => security_group_name})
     else
-      add_flash(_("Unable to create Security Group \"%{name}\": %{details}") % {
+      flash_and_redirect(_("Unable to create Security Group \"%{name}\": %{details}") % {
         :name    => security_group_name,
         :details => task.message
       }, :error)
     end
-
-    @breadcrumbs.pop if @breadcrumbs
-    session[:edit] = nil
```

Comment:
Has this moved somewhere?

If you don't clear `session[:edit]` after finishing the action, how does the controller know we're no longer editing?

Reply:
Yes, exactly. It is already in `flash_and_redirect`.